shell/platform/android/io/flutter/embedding/engine/renderer/SurfaceTextureSurfaceProducer.java

```diff
@@ -0,0 +1,59 @@
+package io.flutter.embedding.engine.renderer;
+
+import android.graphics.SurfaceTexture;
+import android.view.Surface;
+import androidx.annotation.NonNull;
+import io.flutter.view.TextureRegistry;
+
+/** Uses a {@link android.graphics.SurfaceTexture} to populate the texture registry. */
+final class SurfaceTextureSurfaceProducer
+    implements TextureRegistry.SurfaceProducer, TextureRegistry.GLTextureConsumer {
+  private final long id;
+  private int requestBufferWidth;
+  private int requestedBufferHeight;
+
+  @NonNull private final SurfaceTexture texture;
+
+  SurfaceTextureSurfaceProducer(long id) {
+    this.id = id;
+    this.texture = new SurfaceTexture(0);
+  }
+
+  @Override
+  public long id() {
+    return id;
+  }
+
+  @Override
+  public void release() {
+    texture.release();
+  }
+
+  @Override
+  @NonNull
+  public SurfaceTexture getSurfaceTexture() {
+    return texture;
+  }
+
+  @Override
+  public void setSize(int width, int height) {
+    requestBufferWidth = width;
+    requestedBufferHeight = height;
+    getSurfaceTexture().setDefaultBufferSize(width, height);
+  }
+
+  @Override
+  public int getWidth() {
+    return requestBufferWidth;
+  }
+
+  @Override
+  public int getHeight() {
+    return requestedBufferHeight;
+  }
+
+  @Override
+  public Surface getSurface() {
+    return new Surface(getSurfaceTexture());
```

Comment:
You should create the Surface once and cache it.

Reply:
Ack! I was confused by this comment in the parent interface:

```java
/**
 * Get a Surface that can be used to update the texture contents.
 *
 * <p>NOTE: You should not cache the returned surface but instead invoke getSurface each time
 * you need to draw. The surface may change when the texture is resized or has its format
 * changed.
 *
 * @return a Surface to use for a drawing target for various APIs.
 */
Surface getSurface();
```

Perhaps we could modify to specify as a _consumer_ of this API (versus implementor)?

Reply:
For SurfaceTexture you should be able to create one surface and always return that. ImageReader requires we make new Surfaces so our API *users* shouldn't cache the returned Surface.

Reply:
Got it, thanks for clarifying. Done.